web/src/portico/landing-page.ts

```diff
@@ -242,6 +242,14 @@ $(document).on("click", ".markdown h1, .markdown h2, .markdown h3", function ()
     window.location.hash = $(this).attr("id")!;
 });
 
+$(document).on("click", ".nav-zulip-logo", (e) => {
+    const homepage_path = page_params.development_environment ? "/hello/" : "/";
```

Comment:
I think a better check would be of existance of `portico-hello-page` class in the page to check if it is the home page.